docs/graphx-programming-guide.md

```diff
@@ -708,9 +708,8 @@ messages remaining.
 > messaging function.  These constraints allow additional optimization within GraphX.
 
 The following is the type signature of the [Pregel operator][GraphOps.pregel] as well as a *sketch*
-of its implementation (note: to avoid stackOverflowError due to long lineage chains, graph and 
-messages are periodically checkpoint and the checkpoint interval is set by
-"spark.graphx.pregel.checkpointInterval", it can be disable by set as -1):
+of its implementation (note: to avoid stackOverflowError due to long lineage chains, pregel support periodcally
+checkpoint graph and messages by setting "spark.graphx.pregel.checkpointInterval"):
```

Comment:
I think we can recommend a good value (say 10 was the earlier default) to set to since now it defaults to off
Also, good point about checkpointdir - would be good to mention that is required be set as well and link to any doc we have on that